### PR TITLE
Bugfix/155 condense conditions

### PR DESCRIPTION
**Pull request by sappenin**
* Remove `Condition.java` and `Fulfillment.java` in-favor of `InterledgerCondition.java` and `InterledgerFulfillment.java`
* Update all references
* Update unit tests.

### CODE REVIEW

File: ilp-core/src/main/java/org/interledger/core/InterledgerCondition.java [hunk 1]
```diff
@@ -24,68 +24,74 @@
...
    *
-   * @param bytes 32-byte SHA-256 hash digest
-   * @return the condition containing these bytes
+   * @return The {@link InterledgerCondition} containing the supplied hash.
    */
-  static InterledgerCondition from(byte[] bytes) {
-    Objects.requireNonNull(bytes, "bytes cannot be null");
-    return new ImmutableInterledgerCondition(bytes);
+  static InterledgerCondition from(final byte[] hashBytes) {
+    Objects.requireNonNull(hashBytes, "hash cannot be null");
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerCondition.java]: hash vs hashBytes

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java [hunk 2]
```diff
@@ -26,99 +26,103 @@
...
- * <p>This is a wrapper around the 32 byte pre-image from a SHA-256 hash digest.
+ * <p>This is a wrapper around the 32 byte pre-image of a SHA-256 hash digest.</p>
  *
  * @see InterledgerCondition
  */
 public interface InterledgerFulfillment extends Comparable<InterledgerFulfillment> {
 
   /**
-   * Create a new immutable InterledgerFulfillment from the provided 32-bytes.
+   * Create a new immutable InterledgerFulfillment from the provided 32-hash.
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: `32-hash`

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java [hunk 3]
```diff
@@ -26,99 +26,103 @@
...
    */
   static InterledgerFulfillment from(byte[] bytes) {
-    Objects.requireNonNull(bytes, "bytes cannot be null");
+    Objects.requireNonNull(bytes, "hash cannot be null");
     return new ImmutableInterledgerFulfillment(bytes);
   }
 
   /**
-   * Get the internal bytes from the fulfillment.
+   * <p>Get the internal hash from the fulfillment.</p>
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: `hash` vs `preImage ` ?

File: ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java [hunk 4]
```diff
@@ -74,20 +74,20 @@
         {InterledgerPreparePacket.builder()
             .destination(InterledgerAddress.of("test3.foo.bar"))
```
Comment (sublimator) [inline ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: `Fulfillment.of -> InterledgerFulfillment.from`
`Condition.of -> InterledgerCondition.from`

`InterledgerAddress.of -> InterledgerAddress.of` 

?

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java (hunk 2 above)
Comment (sublimator) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: It's a pre-image, not a hash right?

Comment (adrianhopebailie) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: Yep. Looks like a find-replace mistake.

File: ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java (hunk 4 above)
Comment (adrianhopebailie) [reply ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: At first glance I think these do different things. `.of` is a constructor from simple input and `.from` is a clone.

**Comment by sublimator** [comment]
Aaaaah, I saw some changes, just figured one was missed.

(On mobile atm, so can’t see the code)

Of and From are java conventions?

File: ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java (hunk 4 above)
Comment (sublimator) [reply ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: @adrianhopebailie 
Hrmmm, seems using from for `simple inputs` too: 
e.g.
InterledgerCondition.from(conditionBytes)

Comment (adrianhopebailie) [reply ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: We did try to have a convention but it's possible things have gotten mixed up with time.

@sappenin WDYT?

File: ilp-core/src/main/java/org/interledger/core/InterledgerCondition.java (hunk 1 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerCondition.java]: Nic catch! Fixed by 17bf18ea2f212a06aa3b73a9db69ddd1ced7b051

File: ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java (hunk 4 above)
Comment (sappenin) [reply ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: Yeah, there are some typos in the code. Should be `.of` for construction with inputs, and `.from` for copy-constructor-like behavior. I'll fix the code.

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java (hunk 2 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: Fixed by c387f11431c2f1904d8b5bf3729568e3f95959b6

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java (hunk 3 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: Fixed by c387f11

File: ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java (hunk 4 above)
Comment (sappenin) [reply ilp-core-codecs/src/test/java/org/interledger/core/asn/codecs/InterledgerPreparePacketOerSerializerTests.java]: Fixed in 8f8b44d9e9bfc2df359a668d75548dbb03b6a0e2 and 552a691651a658126c85f1bc0d14487c04b5a0a2

**Comment by sappenin** [comment]
@sublimator I think I addressed all comments. Take one more look? Typos and nits welcome - we want to deliver clean code here.

File: jackson-datatypes/pom.xml [hunk 5]
```diff
@@ -8,7 +8,6 @@
     <version>0.12.0-SNAPSHOT</version>
   </parent>
   <modelVersion>4.0.0</modelVersion>
-  <packaging>pom</packaging>
```
Comment (sappenin) [inline jackson-datatypes/pom.xml]: This should fix #156

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java [hunk 6]
```diff
@@ -26,99 +26,103 @@
...
-    private InterledgerCondition condition;
-    private byte[] conditionBytes;
-
-    protected ImmutableInterledgerFulfillment(byte[] bytes) {
-      if (bytes.length != 32) {
-        throw new IllegalArgumentException("InterledgerFulfillment must be exactly 32 bytes.");
+    protected ImmutableInterledgerFulfillment(byte[] preimageBytes) {
+      if (preimageBytes.length != 32) {
+        throw new IllegalArgumentException(
+            "Interledger preimages must be exactly 32 preimageBytes.");
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: > must be exactly 32 preimageBytes.

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java [hunk 7]
```diff
@@ -152,18 +156,18 @@ public int compareTo(InterledgerFulfillment other) {
         return 0;
       }
 
-      //Only call getBytes() if we have to (avoid array copy)
+      //Only call getHash() if we have to (avoid array copy)
```
Comment (sublimator) [inline ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: //Only call getHash()

 //Only call getPreimage()

File: ilp-core/src/test/java/org/interledger/core/InterledgerConditionTest.java [hunk 8]
```diff
@@ -62,39 +63,55 @@
 
   @Test
   public void testGetBytes() {
-    assertArrayEquals(InterledgerCondition.from(BYTES_1).getBytes(), BYTES_1);
+    assertArrayEquals(InterledgerCondition.of(BYTES_1).getHash(), BYTES_1);
```
Comment (sublimator) [inline ilp-core/src/test/java/org/interledger/core/InterledgerConditionTest.java]: testGetHash ?

**Comment by sublimator** [comment]
@sappenin 

Seems good

**Comment by sublimator** [comment]
@adrianhopebailie 
Got your invite. I'm happy to help with clicking merge buttons in the interest of momentum, but I'm not really familiar enough yet with ILP to help much in the way of reviews beyond trivial/cosmetic stuff like typos. 

This one seems good to go though right?

**Comment by sappenin** [comment]
@sublimator let me address a few more changes and I’ll merge this tomorrow. Our custom thus far on the project is to allow (force?) the creator of a PR to do the final merge since that’s usually the person who knows when it’s finalized.

**Comment by sublimator** [comment]
k

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java (hunk 6 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: This is fixed.

File: ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java (hunk 7 above)
Comment (sappenin) [reply ilp-core/src/main/java/org/interledger/core/InterledgerFulfillment.java]: This is fixed.

File: ilp-core/src/test/java/org/interledger/core/InterledgerConditionTest.java (hunk 8 above)
Comment (sappenin) [reply ilp-core/src/test/java/org/interledger/core/InterledgerConditionTest.java]: This is fixed.

**Comment by adrianhopebailie** [comment]
> Got your invite. I'm happy to help with clicking merge buttons in the interest of momentum, but I'm not really familiar enough yet with ILP to help much in the way of reviews beyond trivial/cosmetic stuff like typos.

@sublimator - you should be able to at least create branches now if you want to submit any PRs

**Comment by sublimator** [comment]
Noted!

**Pull request merged**